x-pack/packages/kbn-elastic-assistant/impl/assistant/api/conversations/conversations.ts

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { OpenAiProviderType } from '@kbn/stack-connectors-plugin/common/openai/constants';
+import { HttpSetup } from '@kbn/core/public';
+import { IHttpFetchError } from '@kbn/core-http-browser';
+import {
+  ELASTIC_AI_ASSISTANT_CONVERSATIONS_URL,
+  ELASTIC_AI_ASSISTANT_API_CURRENT_VERSION,
+} from '@kbn/elastic-assistant-common';
+import { Conversation, Message } from '../../../assistant_context/types';
+
+export interface GetConversationByIdParams {
+  http: HttpSetup;
+  id: string;
+  signal?: AbortSignal | undefined;
+}
+
+/**
+ * API call for getting conversation by id.
+ *
+ * @param {Object} options - The options object.
+ * @param {HttpSetup} options.http - HttpSetup
+ * @param {string} options.id - Conversation id.
+ * @param {AbortSignal} [options.signal] - AbortSignal
+ *
+ * @returns {Promise<Conversation | IHttpFetchError>}
+ */
+export const getConversationById = async ({
+  http,
+  id,
+  signal,
+}: GetConversationByIdParams): Promise<Conversation | IHttpFetchError> => {
+  try {
+    const response = await http.fetch(`${ELASTIC_AI_ASSISTANT_CONVERSATIONS_URL}/${id}`, {
+      method: 'GET',
+      version: ELASTIC_AI_ASSISTANT_API_CURRENT_VERSION,
+      signal,
+    });
+
+    return response as Conversation;
+  } catch (error) {
+    return error as IHttpFetchError;
+  }
+};
+
+export interface PostConversationParams {
+  http: HttpSetup;
+  conversation: Conversation;
+  signal?: AbortSignal | undefined;
+}
+
+/**
+ * API call for setting up the Conversation.
+ *
+ * @param {Object} options - The options object.
+ * @param {HttpSetup} options.http - HttpSetup
+ * @param {Conversation} [options.conversation] - Conversation to be added
+ * @param {AbortSignal} [options.signal] - AbortSignal
+ *
+ * @returns {Promise<PostConversationResponse | IHttpFetchError>}
+ */
+export const createConversationApi = async ({
```

Comment:
nit: Consider removing the `Api` in `createConversationApi`, `deleteConversationApi`, `updateConversationApi`